WidgetExtension/Now Playing/NowPlayingEntryView.swift

```diff
@@ -18,39 +19,96 @@ struct NowPlayingWidgetEntryView: View {
 
     var body: some View {
         if let playingEpisode = entry.episode {
-            ZStack {
-                if showsWidgetBackground {
-                    Rectangle().fill(widgetColorScheme.bottomBackgroundColor)
+            switch family {
+            case .systemSmall:
+                smallWidget(playingEpisode: playingEpisode)
+            default:
+                mediumWidget(playingEpisode: playingEpisode)
+            }
+        }
+        else if !showsWidgetBackground {
+            nothingPlaying
+        }
+        else {
+            switch family {
+            case .systemSmall:
+                ZStack {
+                    Image(CommonWidgetHelper.loadAppIconName())
+                        .resizable()
                 }
+                .widgetURL(URL(string: "pktc://last_opened"))
+                .clearBackground()
+            default:
+                nothingPlayingMedium
+            }
+        }
+    }
+
+    private func smallWidget(playingEpisode: WidgetEpisode) -> some View {
+        ZStack {
+            if showsWidgetBackground {
+                Rectangle().fill(widgetColorScheme.bottomBackgroundColor)
+            }
+            VStack(alignment: .leading, spacing: 10) {
+                smallArtwork(playingEpisode: playingEpisode)
+
+                episodeTitle(playingEpisode: playingEpisode)
+
+                playToggleOrPlaybackLabel(playingEpisode: playingEpisode)
+            }
+            .widgetURL(URL(string: "pktc://last_opened"))
+            .clearBackground()
+            .if(!showsWidgetBackground) { view in
+                view
+                    .padding(.top)
+                    .padding(.bottom)
+            }
+        }
+    }
+
+    private func mediumWidget(playingEpisode: WidgetEpisode) -> some View {
+        ZStack {
+            if showsWidgetBackground {
+                Rectangle().fill(widgetColorScheme.bottomBackgroundColor)
+            }
+
+            HStack {
+                LargeArtworkView(imageData: playingEpisode.imageData, size: .infinity)
+
                 VStack(alignment: .leading, spacing: 10) {
-                    artwork(playingEpisode: playingEpisode)
+                    HStack(alignment: .bottom) {
+                        Text("Now Playing")
+                            .font(.caption)
+                            .textCase(.uppercase)
+                            .padding(topPadding)
+                            .foregroundColor(widgetColorScheme.bottomTextColor.opacity(0.6))
+                        Spacer()
+                        Image(widgetColorScheme.iconAssetName)
+                            .frame(width: 28, height: 28)
```

Comment:
It seems like this icon size is used three times, could it be put into a Constants or some other local property?

Reply:
Good call. Wasn't sure where best to add it but I found a `CommonWidgetHelper` class to add the property to, and found all cases where widget icon frame was being defined (it was more than 3) and updated them all.